esp_port/README.md

```diff
@@ -0,0 +1,203 @@
+- `signaling_only`: special application discussed [below](#signalling-and-streaming-split-on-esp32-p4-and-esp32-c6). <br>
+    Signalling only binary flashed on network co-processor, i.e. ESP32-C6
+- `streaming_only`: special app discussed [below](#signalling-and-streaming-split-on-esp32-p4-and-esp32-c6).<br>
+    Streaming only binary flashed on main processor, i.e. ESP32-P4
+
+### Signalling and Streaming Split on ESP32-P4 and ESP32-C6
+- Espressif's innovative solution, `esp_hosted` allows us to run network stack on both the co-processor and host. Network stack running on both the processors, share same IP but different, non-overlapping port numbers.
+- `ESP32-P4 Function_EV_Board` is equipped with on-board main processor, `ESP32-P4` and on-board network co-processor, `ESP32-C6`.
+- Signalling functionality of WebRTC would run on network co-processor, i.e., `ESP32-C6`.
+- Once the streaming request is received by `ESP32-C6`, `ESP32-P4` is notified to start the streaming.
+- This gives us two advantages:
+  - Significant power savings <br>
+  ESP32-P4 is a powerful and fast MCU. But at the same time, consumes higher power.
+  By default, Keeping `ESP32-P4` in `deep sleep` mode and only wake-up once streaming request is received at `ESP32-C6`, would save a lot of power.
+  - Instant wake-up times <br>
+  Since network stack is running on both the sides, ESP32-P4 can instantly get IP address after waking up.
+
+### Using the `split_mode`
+- Similar to `classic_mode` on ESP32-P4, this can be built and flashed.
+  - Build the `streaming_only` application and flash it on ESP32-P4
+  - Build the `signalling_only` application and flash it on ESP32-C6
+  - That's it, you can now start viewer from the [WebRTC Test webpage](https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/index.html).
+    - You can observe that, the ESP32-C6 is connected to the signalling server and relays only absolutely necessary messages to ESP32-P4. Also, it forwards messages from ESP32-P4 to the signalling server.
+    - The final RTP and SCTP sessions are directly established from the ESP32-P4.
+- Please follow [streaming_only](examples/streaming_only/README.md) and [signalling_only](examples/signalling_only/README.md) for more in-depth understanding of how this works.
+
+
+## BUILD
+
+### Configure the project
+Go to the example directory and select the target using following command:
+
+```bash
+# The non-split legacy webrtc app
+cd examples/webrtc_classic
+
+# for esp32
+idf.py set-target esp32
+
+# for esp32s3
+idf.py set-target esp32s3
+
+# for esp32p4
+idf.py set-target esp32p4
+```
+
+Use menuconfig of ESP-IDF to configure the project.
+
+```bash
+idf.py menuconfig
+```
+
+- These parameters under Example Configuration Options must be set:
+
+  - ESP_WIFI_SSID
+  - ESP_WIFI_PASSWORD
+  - ESP_MAXIMUM_RETRY
+  - AWS_ACCESS_KEY_ID
+  - AWS_SECRET_ACCESS_KEY
+  - AWS_DEFAULT_REGION
+  - AWS_KVS_CHANNEL
+  - AWS_KVS_LOG_LEVEL
```

Comment:
I saw some sample frames in one of the folders, would it make sense to have an option to use the camera sensor vs. pre-generated frames?